### PR TITLE
tracking_performance: enable running over campaigns

### PR DESCRIPTION
**Pull request by veprbl**
This enables running over campaigns. The 15 GeV point had to be replaced with 20 GeV because that is what is currently available in the productions.

The current strategy is to run over a single file and enable caching of those to scratch on eicweb. We can expand once this is confirmed to be workable.

### CODE REVIEW

**Comment by Simple-Shyam** [comment]
Thanks, @veprbl. We can also add DCAT (also DCAz, I will commit) to the snake file and then directly use these plots for the TDR with a polishing code attached, which reads the output root file of the performances.


```
void draw_req_Mom(double etamin, double etamax, double xmin=0., double xmax=0.);
void draw_mom(TString particle = "pi-", double etamin=-1.0, double etamax=1.0){
   gStyle->SetPalette(1);
   gStyle->SetOptTitle(1);
   gStyle->SetTitleOffset(1.0,"XY");
   gStyle->SetTitleSize(.04,"XY");
   gStyle->SetLabelSize(.04,"XY");
   gStyle->SetHistLineWidth(2);
   gStyle->SetOptFit(1);
   gStyle->SetOptStat(1);

  TCanvas *c_mom = new TCanvas("cmom","cmom",1400,1000);
  c_mom->SetMargin(0.10, 0.05 ,0.1,0.05);
  c_mom->SetGrid();

  TFile *f = TFile::Open(Form("mom_resol_%1.1f_eta_%1.1f.root",etamin,etamax));
  TMultiGraph *mg = (TMultiGraph*)f->Get(Form("mom_resol_%1.1f_eta_%1.1f",etamin,etamax));
  TGraphErrors *gr = (TGraphErrors*) mg->GetListOfGraphs()->At(1);
  c_mom->cd();
  gr->Draw("AP");
  gr->GetXaxis()->SetRangeUser(0.45,15.2);
  gr->GetYaxis()->SetRangeUser(0.,2.0*TMath::MaxElement(gr->GetN(),gr->GetY()));
  draw_req_Mom(etamin,etamax,0.,gr->GetXaxis()->GetXmax());
  
  TLegend *lmom = new TLegend(0.70,0.80,0.90,0.93);
  lmom->SetTextSize(0.03);
  lmom->SetBorderSize(0);
  lmom->SetHeader(Form("Particle (%s): %1.1f < #eta < %1.1f",particle.Data(),etamin,etamax),"C");
  lmom->AddEntry(gr,"Real Seed");
  lmom->Draw("same");
}

void draw_req_Mom(double etamin, double etamax, double xmin=0., double xmax=0.)
{

   TF1 *dd4hep_p;
   if (etamin >= -3.5 && etamax <= -2.5) dd4hep_p = new TF1("dd4hep_p", "TMath::Sqrt((0.1*x)^2+2.0^2)",xmin,xmax);
   else if (etamin >= -2.5 && etamax <= -1.0) dd4hep_p = new TF1("dd4hep_p", "TMath::Sqrt((0.05*x)^2+1.0^2)",xmin,xmax);
   else if (etamin >= -1.0 && etamax <= 1.0) dd4hep_p = new TF1("dd4hep_p", "TMath::Sqrt((0.05*x)^2+0.5^2)",xmin,xmax);
   else if (etamin >= 1.0 && etamax <= 2.5) dd4hep_p = new TF1("dd4hep_p", "TMath::Sqrt((0.05*x)^2+1.0^2)",xmin,xmax);
   else if (etamin >= 2.5 && etamax <= 3.5) dd4hep_p = new TF1("dd4hep_p", "TMath::Sqrt((0.1*x)^2+2.0^2)",xmin,xmax);
   else return;
   dd4hep_p->SetLineStyle(7);
   dd4hep_p->SetLineColor(kMagenta);
   dd4hep_p->SetLineWidth(3.0);
   dd4hep_p->Draw("same");

  TLegend *l= new TLegend(0.70,0.75,0.90,0.80);
  l->SetTextSize(0.03);
  l->SetBorderSize(0);
  l->AddEntry(dd4hep_p,"PWGReq","l");
  l->Draw("same");
 }
 ```

**Comment by veprbl** [comment]
@Simple-Shyam Of course! Please add those changes to #34.

**Comment by Simple-Shyam** [comment]
> @Simple-Shyam Of course! Please add those changes to #34.

I will do it.

**Comment by veprbl** [comment]
This is ready, I believe.

**Comment by Simple-Shyam** [comment]
Thanks, I will check the code and merge. I have presentation on this Friday in collaboration meeting. I will do it on weekend, if it's fine.  I didn't see the artifact plots (I guess it's not ready yet)?  
https://eicweb.phy.anl.gov/EIC/benchmarks/detector_benchmarks/-/jobs/3541246/artifacts/browse/results/

**Comment by veprbl** [comment]
> Thanks, I will check the code and merge. I have presentation on this Friday in collaboration meeting. I will do it on weekend, if it's fine. I didn't see the artifact plots (I guess it's not ready yet)? https://eicweb.phy.anl.gov/EIC/benchmarks/detector_benchmarks/-/jobs/3541246/artifacts/browse/results/

That link is for `sim:tracking_performance`, which is a simulation step before running the local benchmark in `bench:tracking_performance`.
The campaign plots you can find in `bench:tracking _performance_campaigns` or `collect_results:tracking _performance_campaigns`.

### 23.10.0, $1 < \eta < 2.5$

### 23.11.0, $1 < \eta < 2.5$

### 23.12.0, $1 < \eta < 2.5$

### 24.03.1, $1 < \eta < 2.5$

### 24.04.0, $1 < \eta < 2.5$

**Comment by veprbl** [comment]
https://eicweb.phy.anl.gov/EIC/benchmarks/detector_benchmarks/-/jobs/3541344/artifacts/browse

**Comment by Simple-Shyam** [comment]
Hi Dmitry, I checked the plots and I found that the ePIC (epic software) version is missing however I can see EICRecon version there. Once that's fixed. I will merge it. I also noticed that it's not setting automatic range on y axis which I pushed in the morning in which the maximum value on y-axis will be 50 % more than the maximum value. 

https://eicweb.phy.anl.gov/EIC/benchmarks/detector_benchmarks/-/jobs/3564682/artifacts/browse/results/tracking_performances/

File: benchmarks/tracking_performances/doCompare_truth_real_widebins_mom.C [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #define mpi 0.139  // 1.864 GeV/c^2
 
 void draw_req_Mom(double etamin, double etamax, double xmin=0., double xmax=0.);
-void doCompare_truth_real_widebins_mom(TString particle = "pi-",double etamin=-1.0, double etamax=1.0, double range =0.3, Bool_t drawreq=1, TString epic ="", TString eicrecon = "") // name = p, pt for getting p or pt dependence fitted results
+void doCompare_truth_real_widebins_mom(TString particle = "pi-",double etamin=-1.0, double etamax=1.0, double range =0.3, Bool_t drawreq=1, TString extra_legend = "") // name = p, pt for getting p or pt dependence fitted results
```
Comment (Simple-Shyam) [inline benchmarks/tracking_performances/doCompare_truth_real_widebins_mom.C:13]: Can I know why you removed the ePIC and EICRecon versions?

Comment (veprbl) [reply benchmarks/tracking_performances/doCompare_truth_real_widebins_mom.C:13]: They are not removed, the whole label is defined in Snakemake. The point is that we only want to specify the campaign tag for official production (the other versions are fixed and known in that case).

Comment (Simple-Shyam) [reply benchmarks/tracking_performances/doCompare_truth_real_widebins_mom.C:13]: I see that performances are degraded again. What has changed now? Please check the plot below. If we tag the plot with the version, we will know which version is giving the correct result. It looks again the issue of the material map.
https://eicweb.phy.anl.gov/EIC/benchmarks/detector_benchmarks/-/jobs/3572750/artifacts/file/results/tracking_performances/mom_resol_1.0_eta_2.5.png

Comment (veprbl) [reply benchmarks/tracking_performances/doCompare_truth_real_widebins_mom.C:13]: What you show is not the latest, or not from this PR. The latest job's plot is
https://eicweb.phy.anl.gov/EIC/benchmarks/detector_benchmarks/-/jobs/3571493/artifacts/file/results/tracking_performances/local/mom_resol_1.0_eta_2.5.png

Comment (Simple-Shyam) [reply benchmarks/tracking_performances/doCompare_truth_real_widebins_mom.C:13]: Thanks

**Pull request merged**